CoreFindView/src/au/gov/asd/tac/constellation/views/find2/plugins/advanced/AdvancedSearchPlugin.java

```diff
@@ -0,0 +1,717 @@
+                final int currElement = GraphElementType.VERTEX.getElement(graph, i);
+                graph.setBooleanValue(nodeSelectedAttribute, currElement, false);
+            }
+        }
+        // loop through all transactions that are selected and deselect them
+        if (transactionSelectedAttribute != Graph.NOT_FOUND) {
+            for (int i = 0; i < transactionsCount; i++) {
+                final int currElement = GraphElementType.TRANSACTION.getElement(graph, i);
+                graph.setBooleanValue(transactionSelectedAttribute, currElement, false);
+            }
+        }
+    }
+
+    @Override
+    protected void edit(GraphWriteMethods graph, PluginInteraction interaction, PluginParameters parameters) throws InterruptedException, PluginException {
+        //Retrieve the existing FindResultList Meta attribute
+        final int stateId = FindViewConcept.MetaAttribute.FINDVIEW_STATE.ensure(graph);
+        FindResultsList foundResult = graph.getObjectValue(stateId, 0);
+
+        /**
+         * If it doesn't exist or is null, create a new list with the starting
+         * index and the current find parameters. If it does exist, create a
+         * list with the correct index and the current find parameters
+         */
+        if (foundResult == null) {
+            foundResult = new FindResultsList(STARTING_INDEX, this.parameters, graph.getId());
+        } else {
+            /**
+             * This is delicate, so don't change. This process, captures the
+             * users previous search and their current search, compares the 2 to
+             * see if they are the same. If yes then get the previous index of
+             * the last foundResult. If its different reset the index. The
+             * parameters are instantiated as new variables as there were
+             * manipulation issues elsewhere causing this process to fail.
+             */
+            final FindResultsList oldList = new FindResultsList(STARTING_INDEX, foundResult.getAdvancedSearchParameters(), foundResult.getGraphId());
+            final AdvancedSearchParameters oldparameters = oldList.getAdvancedSearchParameters();
+            final AdvancedSearchParameters newParamters = new AdvancedSearchParameters(this.parameters);
+            int newIndex = getIndex(newParamters, oldparameters, foundResult.getCurrentIndex());
+            foundResult = new FindResultsList(newIndex, newParamters, oldList.getGraphId());
+        }
+
+        foundResult.clear();
+        graph.setObjectValue(stateId, 0, foundResult);
+
+        boolean found;
+        final int elementCount = elementType.getElementCount(graph);
+
+        // do this if add to selection
+        if (currentSelection.equals("Ignore")) {
+            clearSelection(graph);
+        }
+        findInCurrentSelectionList = new FindResultsList(graph.getId());
+        FindResultsList findAllMatchingResultsList = new FindResultsList(graph.getId());
```

Comment:
final. There's a few finals missing in this function but I won't be through all of them